src/Colopl/Spanner/Concerns/ManagesSessionPool.php

```diff
@@ -81,7 +81,8 @@ public function listSessions(): Collection
     {
         $databaseName = $this->getSpannerDatabase()->name();
         $response = (new ProtobufSpannerClient())->listSessions($databaseName);
-        return collect($response->iterateAllElements())->map(function (ProtobufSpannerSession $session) {
+        return collect($response->iterateAllElements())->map(function ($session) {
+            assert($session instanceof ProtobufSpannerSession);
```

Comment:
Why was this changed to an assertion?

Reply:
Hi @taka-oyama 
I got phpstan error there (probably because phpstan version not fixed in composer.json it installed the most recent version of phpstan - you probably will get the same errors if delete composer.lock and rerun tests/install) 
```
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Colopl/Spanner/Concerns/ManagesSessionPool.php (in context of class Colopl\Spanner\Connection)                                                            
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------- 
  84     Parameter #1 $callback of method Illuminate\Support\Collection<(int|string),mixed>::map() expects callable(mixed, (int|string)): Colopl\Spanner\Session,  
         Closure(Google\Cloud\Spanner\V1\Session): Colopl\Spanner\Session given.                                                                                   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------- 
```
and when I removed ProtobufSpannerSession then error changed to 
```
 ------ --------------------------------------------------------------------------------------------------------------------------------- 
  Line   Colopl/Spanner/Concerns/ManagesSessionPool.php (in context of class Colopl\Spanner\Connection)                                   
 ------ --------------------------------------------------------------------------------------------------------------------------------- 
  85     Parameter #1 $protobufSession of class Colopl\Spanner\Session constructor expects Google\Cloud\Spanner\V1\Session, mixed given.  
 ------ --------------------------------------------------------------------------------------------------------------------------------- 

```

and according to https://phpstan.org/writing-php-code/narrowing-types
the only suggested way - explicit checking of types - via assert or directly

Reply:
I'm starting to think phpstan level 9 might be too high for laravel based packages.
Would it bother you if I lowered it to 8? In which case I think this error will not appear here.

Reply:
I just found out that asserts can be tuned to not have performance impact on production so this code is OK.

https://www.php.net/manual/en/function.assert.php#function.assert.expectations